common/ferrostar/src/routing_adapters/valhalla.rs

```diff
@@ -30,32 +43,41 @@ pub struct ValhallaHttpRequestGenerator {
     /// The Valhalla costing model to use.
     profile: String,
     // TODO: Language, units, and other top-level parameters
-    /// JSON costing options to pass through.
-    costing_options: JsonValue,
+    /// JSON arbitrary key/values which override the defaults.
+    ///
+    /// These may contain nested keys.
+    options: JsonValue,
 }
 
 impl ValhallaHttpRequestGenerator {
-    pub fn new(endpoint_url: String, profile: String, costing_options: Option<JsonValue>) -> Self {
+    /// Creates a new Valhalla request generator given an endpoint URL, a profile name,
+    /// and options (which will update the minimal defaults set internally).
+    ///
+    /// NOTE: If options is `None` or a non-object, it will be interpreted as an empty object.
+    pub fn new(endpoint_url: String, profile: String, options: Option<JsonValue>) -> Self {
         Self {
             endpoint_url,
             profile,
-            costing_options: costing_options.unwrap_or(json!({})),
+            options: options.unwrap_or(json!({})),
         }
     }
 
-    pub fn with_costing_options_json(
+    /// Creates a new Valhalla request generator given an endpoint URL, a profile name,
+    /// and options (which will update the minimal defaults set internally).
+    /// NOTE: If options is `None` or a non-object, it will be interpreted as an empty object.
```

Comment:
"it will be interpreted as an empty object." I would suggest we describe how this impacts the minimal defaults set internally. This phrasing makes it hard to understand how `None` impacts the internal defaults. E.g is None just the defaults or is None actually no options at all?

Reply:
Excellent point. I just pushed  a revision which I think clarifies things (there is no effect; it's just ignored).

Reply:
Now that you mention in though.... maybe we can do better... and make it fail in that case since that's a bit surprising. What we had before wasn't very obvious either.

Reply:
Ok; that's better. Should be all cleaned up. It isn't something that needs explaining now ;) The constructors now either make illegal state unrepresentable or return result types.